### PR TITLE
feat: complete dashboard page

### PR DESCRIPTION
**Pull request by manuel-calavera**
what changed:

1. complete dashboard page with all components in place (header, stats, map, chart, footer, etc). only links are missing for now.
2. enable socket connection and api 
3. lots of layout and styles refactoring 
4. routed to /rc (release candidate) for now

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -4,12 +4,12 @@ import {BrowserRouter} from 'react-router-dom';
...
 import './index.css';
 import App from './App';
-// import AppV2 from './AppV2';
+import AppV2 from './AppV2';
 import * as serviceWorker from './serviceWorker';
 
 ReactDOM.render(
   <BrowserRouter>
-    {window.location.pathname.includes('v2') ? <App /> : <App />}
+    {window.location.pathname.includes('rc') ? <AppV2 /> : <App />}
```
Comment (mvines) [inline src/index.js]: Why the change from v2 to rc?

Comment (manuel-calavera) [reply src/index.js]: per our discussion on slack:

@mvines for routing question. it is ok if @sunny continue to develop his own components, i find it useful as a reference. i check them and learn how data is being requested and parsed. i believe it is quite overwork you do on your side but i am fine if you want to do it. it is like intermediate step between mvp and actual production code i am doing.

Michael Vines [12:40 PM]
@Manuel Calavera - between you and @sunny, just do what you both feel is the most efficient.

basically we have conflict on v2 route since @sunnygleason is doing his own react components on the same route.

Comment (mvines) [reply src/index.js]: fine, this is gross but 🙈.  Next week we should try to get on the same non-v1 route if at all possible.  Maintaining 3 routes in the same tree is going to quickly turn into 🍝

Comment (sunnygleason) [reply src/index.js]: 👍 agreed, let's get this sorted

File: src/AppV2.js [hunk 2]
```diff
@@ -5,19 +5,23 @@ import {hot} from 'react-hot-loader/root';
 import {Route, Switch} from 'react-router-dom';
 import Header from 'v2/components/Header';
 import NavBar from 'v2/components/NavBar';
+import Footer from 'v2/components/Footer';
 import theme from 'v2/theme';
+import socket from 'v2/stores/socket';
+
+socket.init();
```
Comment (sunnygleason) [inline src/AppV2.js]: this seems like it needs to turn into an event-based rather than a static call based on @mvines PR for endpoint changes

File: src/v2/components/Dashboard/NetworkOverview/NodesMap/index.jsx [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+        />
+      </MapTooltip>
+    </Marker>
+  );
+  return (
+    <div className={classes.card}>
+      <Typography>Nodes Map</Typography>
+      <ComposableMap
+        projectionConfig={mapConfig.projection}
+        width={390}
```
Comment (sunnygleason) [inline src/v2/components/Dashboard/NetworkOverview/NodesMap/index.jsx]: is this approach specifying width/height responsive?

File: src/v2/components/Dashboard/NetworkOverview/NodesMap/index.jsx [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+            <div className={classes.tooltipDesc}>Gossip: {marker.gossip}</div>
+          </>
+        )}
+      >
+        <circle
+          cx={0}
+          cy={0}
+          r={5}
+          style={{
+            stroke: '#00FFAD',
```
Comment (sunnygleason) [inline src/v2/components/Dashboard/NetworkOverview/NodesMap/index.jsx]: this constant should probably be extracted into theme/config

File: src/v2/components/Dashboard/NetworkOverview/NodesMap/index.jsx [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+        style={mapConfig.style}
+      >
+        <ZoomableGroup center={mapConfig.center} disablePanning>
+          <Geographies
+            geography={`${process.env.PUBLIC_URL}/resources/world-50m-simplified.json`}
+          >
+            {(geographies, projection) =>
+              geographies.map(
+                (geography, i) =>
+                  geography.id !== 'ATA' && (
```
Comment (sunnygleason) [inline src/v2/components/Dashboard/NetworkOverview/NodesMap/index.jsx]: what's ATA, Antarctica?

File: src/v2/components/Dashboard/NetworkOverview/TPS/index.jsx [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+  if (!data) return null;
+  return (
+    <div className={classes.tooltip}>
+      <div className={classes.tooltipTitle}>AVG TPS:{data.y}</div>
+      <div className={classes.tooltipDate}>{data.date}</div>
+    </div>
+  );
+};
+
+const TPS = () => {
```
Comment (sunnygleason) [inline src/v2/components/Dashboard/NetworkOverview/TPS/index.jsx]: This needs a descriptive name that isn't an abbreviation

File: src/v2/stores/networkOverview.js [hunk 7]
```diff
@@ -1,29 +1,114 @@
...
-  {
-    state: observable,
-    globalStats: observable,
-    getStats: action.bound,
-  },
-);
-
-export default OverviewStore;
+
+      this.statsChanges = compose(
```
Comment (sunnygleason) [inline src/v2/stores/networkOverview.js]: I think this should probably be removed until the changes are calculated correctly (daily etc. vs per-message)

File: src/v2/utils/calcChanges.js [hunk 8]
```diff
@@ -0,0 +1,8 @@
+// @flow
+
+export default (oldValue: string | number, newValue: string | number) => {
```
Comment (sunnygleason) [inline src/v2/utils/calcChanges.js]: this needs to be a server-side function (or client-side function that uses the API differently because the desired data isn't present right now)

File: src/v2/stores/socket.js [hunk 9]
```diff
@@ -0,0 +1,22 @@
+import {initSocket} from 'v2/api/socket';
+import networkOverview from 'v2/stores/networkOverview';
+import nodes from 'v2/stores/nodes';
+
+const socketActions = {
```
Comment (sunnygleason) [inline src/v2/stores/socket.js]: this should be refactored a bit because the socket isn't a different store; each message can be retrieved in one of 2 ways: (a) via API for initial population or (b) incoming real-time via the socket. I think the model ends up being similar to initial population of chat history in a chat app

File: src/index.js (hunk 1 above)
Comment (sunny) [reply src/index.js]: Hi there 👋 @manuel-calavera, I don't think I'm the @sunny you were looking for :)
